platforms/core-configuration/configuration-cache/src/main/kotlin/org/gradle/internal/cc/impl/problems/ConfigurationCacheProblems.kt

```diff
@@ -165,6 +165,11 @@ class ConfigurationCacheProblems(
         onProblem(problem, ProblemSeverity.Interrupting)
     }
 
+    override fun onExecutionTimeProblem(problem: PropertyProblem) {
+        val severity = if (isWarningMode) ProblemSeverity.Deferred else ProblemSeverity.Interrupting
+        onProblem(problem, severity)
+    }
```

Comment:
ℹ️  This is the main change for production code. Previously, the execution-time problems were treated as deferred.